ShopTask/Views/Home/CategoriesPartial.cshtml

```
@@ -0,0 +1,18 @@
+﻿@model List<ShopTask.Models.Category>
+
+@{ 
+    int indexOfNewCategory = Model.IndexOf(Model.FirstOrDefault(category => category.Id == null));
```

Чуть короче - newCategoryIndex

---

В данном случае ты знаешь, что айтем под новую позицию всегда должен быть в модели, это часть контракта между view и контроллером. Поэтому вместо Model.FirstOrDefault(category => category.Id == null) лучше использовать Model.Single(category => category.Id == null) - принцип fail fast. Иначе если по какой-то причине контроллер не добавит отдельный айтем под новую, то ошибка может не упасть и сразу будет не очевидно, в чем дело.